test/processImage.js

```diff
@@ -1327,58 +1334,55 @@ describe('express-processimage', () => {
 
   describe('against a real server', () => {
     it('should destroy the created filters when the client closes the connection prematurely', () => {
-      let server;
-      const createdStreams = [];
-      let request;
-      return expect
-        .promise(run => {
-          config.filters = {
-            montage: run(() => ({
-              create: run(() => {
-                const stream = new Stream.Transform();
-                stream._transform = (chunk, encoding, callback) => {
-                  setTimeout(() => {
-                    callback(null, chunk);
-                  }, 1000);
-                };
-                stream.destroy = sandbox.spy().named('destroy');
-                createdStreams.push(stream);
-                setTimeout(
-                  run(() => {
-                    request.abort();
-                  }),
-                  0
-                );
-                return stream;
-              })
-            }))
-          };
-          server = express()
-            .use(processImage(config))
-            .use(express.static(root))
-            .listen(0);
-
-          const serverAddress = server.address();
-          const serverHostname =
-            serverAddress.address === '::'
-              ? 'localhost'
-              : serverAddress.address;
-          const serverUrl = `http://${serverHostname}:${serverAddress.port}/testImage.png?montage`;
-
-          request = http.get(serverUrl);
-          request.end();
-          request.once(
-            'error',
-            run(err => {
-              expect(err, 'to have message', 'socket hang up');
-            })
-          );
+      let resolveOnPipeline;
+      const onPipelinePromise = new Promise(
+        resolve => (resolveOnPipeline = resolve)
+      );
+      const config = {
+        onPipeline: p => {
+          resolveOnPipeline(p);
+        }
+      };
+
+      const server = express()
+        .use(processImage(config))
+        .use(express.static(root))
+        .listen(0);
+      const serverAddress = server.address();
+      const serverHostname =
+        serverAddress.address === '::' ? 'localhost' : serverAddress.address;
+      const serverUrl = `http://${serverHostname}:${serverAddress.port}/testImage.png?progressive`;
+      const request = http.get(serverUrl);
+      request.end();
+
+      const spies = [];
+
+      return onPipelinePromise
+        .then(pipeline => {
```

Comment:
Let's use `async`/`await` here if we're rewriting it anyway

Reply:
Great shout.